lib/vericite.rb

```diff
@@ -0,0 +1,456 @@
+      course = assignment.context
+      today = course.time_zone.today
+      settings = VeriCite::Client.normalize_assignment_vericite_settings(settings)
+
+      response = sendRequest(:create_assignment, settings.delete(:created) ? '3' : '2', settings.merge!({
+        :user => course,
+        :course => course,
+        :assignment => assignment,
+        :utp => '2',
+        :dtstart => "#{today.strftime} 00:00:00",
+        :dtdue => "#{today.strftime} 00:00:00",
+        :dtpost => "#{today.strftime} 00:00:00",
+        :late_accept_flag => '1',
+        :post => true
+      }))
+
+      response.success? ? { assignment_id: response.assignment_id } : response.error_hash
+    end
+
+    # if asset_string is passed in, only submit that attachment
+    def submitPaper(submission, asset_string=nil)
+      student = submission.user
+      assignment = submission.assignment
+      course = assignment.context
+      opts = {
+        :post => true,
+        :utp => '1',
+        :user => student,
+        :course => course,
+        :assignment => assignment,
+        :tem => email(course),
+        :role => submission.grants_right?(student, :grade) ? "Instructor" : "Learner"
+      }
+      responses = {}
+      if submission.submission_type == 'online_upload'
+        attachments = submission.attachments.select{ |a| a.vericiteable? && (asset_string.nil? || a.asset_string == asset_string) }
+        attachments.each do |a|
+          # do not resubmit if the score already exists
+          if submission.vericite_data_hash[a.asset_string][:similarity_score].blank?
+            paper_id = a.id
+            paper_title = File.basename(a.display_name, ".*")
+            paper_ext = a.extension
+            paper_type = a.content_type
+            if paper_ext == nil
+              paper_ext = ""
+            end
+            paper_size = 100 # File.size(
+            responses[a.asset_string] = sendRequest(:submit_paper, '2', { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => a.open(), :ptype => '2' }.merge!(opts))
+          end
+        end
+      elsif submission.submission_type == 'online_text_entry' && (asset_string.nil? || submission.asset_string == asset_string)
+        # do not resubmit if the score already exists
+        if submission.vericite_data_hash[submission.asset_string][:similarity_score].blank?      
+          paper_id = Digest::SHA1.hexdigest submission.plaintext_body
+          paper_ext = "html"
+          paper_title = "InlineSubmission"
+          plain_text = "<html>#{submission.plaintext_body}</html>"
+          paper_type = "text/html"
+          paper_size = plain_text.bytesize
+          
+          responses[submission.asset_string] = sendRequest(:submit_paper, '2', {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => plain_text, :ptype => "1" }.merge!(opts))
```

Comment:
@ccutrer  This looks like its a left over from code that is no longer used. You can see that the ":ptype" argument is never used is vericite.rb

Do I need to create a new PR? This is what the diff looks like:

```
diff --git a/lib/vericite.rb b/lib/vericite.rb
index fd421ba..2a7a7f8 100644
--- a/lib/vericite.rb
+++ b/lib/vericite.rb
@@ -131,12 +131,11 @@ module VeriCite
             paper_id = a.id
             paper_title = File.basename(a.display_name, ".*")
             paper_ext = a.extension
-            paper_type = a.content_type
             if paper_ext == nil
               paper_ext = ""
             end
             paper_size = 100 # File.size(
-            responses[a.asset_string] = sendRequest(:submit_paper, { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => a.open(), :ptype => '2' }.merge!(opts))
+            responses[a.asset_string] = sendRequest(:submit_paper, { :pid => paper_id, :ptl => paper_title, :pext => paper_ext, :psize => paper_size, :pdata => a.open()}.merge!(opts))
           end
         end
       elsif submission.submission_type == 'online_text_entry' && (asset_string.nil? || submission.asset_string == asset_string)
@@ -144,10 +143,9 @@ module VeriCite
         paper_ext = "html"
         paper_title = "InlineSubmission"
         plain_text = "<html>#{submission.plaintext_body}</html>"
-        paper_type = "text/html"
         paper_size = plain_text.bytesize
 
-        responses[submission.asset_string] = sendRequest(:submit_paper, {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :ptype => paper_type, :psize => paper_size, :pdata => plain_text, :ptype => "1" }.merge!(opts))
+        responses[submission.asset_string] = sendRequest(:submit_paper, {:pid => paper_id, :ptl => paper_title, :pext => paper_ext, :psize => paper_size, :pdata => plain_text}.merge!(opts))
       else
         raise "Unsupported submission type for VeriCite integration: #{submission.submission_type}"
       end
```